### PR TITLE
KAFKA-3728 EndToEndAuthorizationTest offsets_topic misconfigured

### PR DESCRIPTION
**Pull request by edoardocomar**
Set OffsetsTopicReplicationFactorProp to 3 like MinInSyncReplicasProp  Else a consumer was able to consume via assign but not via subscribe, so the testProduceAndConsume is now duplicated to check both paths

### CODE REVIEW

**Comment by edoardocomar** [comment]
Hi @ijuma I've implemented your suggestions.

I got rid of the TopicAuthorizationException on __consumer_offsets but 
I could not get rid of OffsetCommitRequiredAcksProp=1

otherwise testProduceConsumeViaSubscribe fails if it's not the first one to be
executed (the test passes on its own).

Disappointing, this may be a cleanup problem but I could not spot any.

**Comment by ijuma** [comment]
@edoardocomar, sorry for the delay. I investigated the issue and I think #1455 should fix it (I briefly tested your PR with it).

**Comment by edoardocomar** [comment]
thanks @ijuma does this mean that I should be able to get rid of 
 OffsetCommitRequiredAcksProp=1 ?

**Comment by ijuma** [comment]
Yes.

**Comment by ijuma** [comment]
@edoardocomar, can you please rebase this now that #1455 was merged?

File: core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala [hunk 1]
```diff
@@ -170,7 +171,22 @@ trait EndToEndAuthorizationTest extends IntegrationTestHarness with SaslSetup {
...
+
+  @Test
+  def testProduceConsumeViaSubscribe {
+    produceProlog()
+    consumers.head.subscribe(List(topic).asJava)
+    consumeRecords(this.consumers.head, numRecords)
+    debug("Finished consuming")
+  }
+
+  private def produceProlog() {
```
Comment (ijuma) [inline core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala]: Maybe call this `setAclsAndProduce()`

**Comment by edoardocomar** [comment]
Thanks @ijuma with your changes in #1455 the only fix needed for this defect was to 
Set OffsetsTopicReplicationFactorProp to 3 like MinInSyncReplicasProp
without which a consumer could not consume vai subscribe.

The other changes are no longer needed

**Comment by ijuma** [comment]
LGTM @edoardocomar. Can you please update the PR description to match the commit message? The PR message is what gets used by our PR merge script in the merge commit.

I'll merge it to trunk and 0.10.0 once you do that and the tests pass. Merging to 0.10.0 to make it easier to backport KAFKA-3396.

**Comment by edoardocomar** [comment]
Hi @ijuma all the tests pass for me - unlike in the Jenkins build. How do I re-kick a Jenkins build?
thanks

**Comment by ijuma** [comment]
I'll run the tests locally. Can you please not repeat the PR title in the description?

**Comment by edoardocomar** [comment]
done thx

**Pull request closed without merging**